orca/internal/internal.go

```diff
@@ -0,0 +1,30 @@
+/*
+ *
+ * Copyright 2022 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+// Package internal contains orca-internal code, for testing purposes and to
+// avoid polluting the godoc of the top-level orca package.
+package internal
+
+import "time"
+
+// MinReportingIntervalForTesting returns the absolute minimum value supported
+// for out-of-band metrics reporting from the ORCA service implementation
+// provided by the orca package.
+//
+// For testing purposes only.
+var MinReportingIntervalForTesting func() time.Duration
```

Comment:
Why change this to a function instead of a `var`?

Reply:
I thought your suggestion was to avoid global vars. Having a function means that the override is used only when the function is set. When the function is not set (which will be the case in all non-test code), we will use the const value.

Because if this was a `var`, how better is it compared to having the `var` in the `internal` package?

Reply:
This is still a global var.  😄

The suggestion was to not make it a global at all, e.g. a hidden entry in `ServiceOptions` to control the behavior per-instance instead of globally.  Something like this:

```go
package orca

type ServiceOptions struct {
	// other fields
	minReportingInterval time.Duration
}
var defaultMinReportingInterval = 30 * time.Second  // use if unset

internal.SetMinReportingInterval = func(so *ServiceOptions, d time.Duration) {
	so.minReportingInterval = d
}

package internal

var internal.SetMinReportingInterval interface{}  // func(*ServiceOptions, time.Duration)
```

Reply:
I thought about this, but this does not provide the user an option to set the `minReportingInterval`. It allows only our tests to do so. We still want to allow the user to set a value greater than the default of `30s`.

Reply:
I don't understand. This is a separate setting that would override the minimum that we allow to be set, which is hard-coded but overridden in tests.

Or make it a simple bool?:

```go
package orca

type ServiceOptions struct {
	// other fields
	allowAnyMinReportingInterval bool
}

internal.AllowAnyMinReportingInterval = func(so *ServiceOptions) {
	so.AllowAnyMinReportingInterval = true
}

package internal

// AllowAnyMinReportingInterval prevents clamping of the min reporting interval configured in the service.
var internal.AllowAnyMinReportingInterval interface{}  // func(*ServiceOptions)
```

(Names TBD.)

Reply:
Done thanks.